presto-kafka07/src/main/java/com/facebook/presto/kafka/KafkaMetadata.java

```diff
@@ -225,6 +227,11 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
         }
 
         log.info("startTs: %s, endTs: %s", startTs, endTs);
+        if (config.isHardLimitOn() && startTs == null && endTs == null) {
```

Comment:
Here is some thoughts about your expectations:
1. If the user doesn't know how to specify a range, they will fall into default query interval.
2. If the user knows how to specify a range, they must agree with default query interval and must have a specific reason to go beyond the default query interval. It would be costly to serve the users if they actually have this specific request. It will involve code review and service deploy processes which wouldn't the ideal way for serving our customers for urgent needs.